web/src/features/dashboard/lib/score-analytics-utils.ts

```diff
@@ -108,8 +108,10 @@ function groupCategoricalScoreDataByTimestamp(
 ): Record<string, DatabaseRow[]> {
   return data.reduce(
     (acc, row) => {
-      const timestamp = row[scoreTimestampAccessor];
-      if (!(timestamp instanceof Date)) return acc;
+      if (row[scoreTimestampAccessor] === null) {
+        return acc;
+      }
+      const timestamp = new Date(row[scoreTimestampAccessor] ?? new Date());
```

Comment:
@marliessophie It seems that clickhouse doesn't return the metadata that trpc needs to auto-cast date columns to a Date type. Instead, I try to cast any string here as long as there is a value set.
Do you have any concerns about this approach? It looks like the function is only used on the dashboard and there it seems to work fine.

Reply:
Clarified offline - change should be good.